collector/generatorreceiver/generator_receiver.go

```diff
@@ -29,25 +29,28 @@ type generatorReceiver struct {
 	server         *httpServer
 }
 
-func (g generatorReceiver) loadTopoFile(topoInline string, path string) (*topology.File, error) {
-	var topoFile topology.File
-
+func (g generatorReceiver) loadTopoFile(topoInline string, path string) (topoFile *topology.File, err error) {
 	// fetch from env var.
 	if len(topoInline) > 0 {
 		g.logger.Info("reading topo inline")
-		err := json.Unmarshal([]byte(topoInline), &topoFile)
+		err = json.Unmarshal([]byte(topoInline), topoFile)
 		if err != nil {
 			return nil, fmt.Errorf("could not parse inline json file: %v", err)
 		}
-		return &topoFile, nil
+	} else {
+		g.logger.Info("reading topo from file path", zap.String("path", g.topoPath))
+		topoFile, err = parseTopoFile(path)
+		if err != nil {
+			return nil, err
+		}
 	}
-	g.logger.Info("reading topo from file path", zap.String("path", g.topoPath))
-	parsedFile, err := parseTopoFile(path)
-
+	err = topoFile.Topology.Load()
 	if err != nil {
 		return nil, err
 	}
-	return parsedFile, nil
+	flags.Manager.LoadFlags(topoFile.Flags, g.logger)
```

Comment:
Refactored this function a little so that we can do all loading that's required prior to validation.